pkg/server/settingswatcher/settings_watcher.go

```diff
@@ -40,14 +42,14 @@ type SettingsWatcher struct {
 	settings *cluster.Settings
 	f        *rangefeed.Factory
 	stopper  *stop.Stopper
-	dec      RowDecoder
 	storage  Storage
 
 	overridesMonitor OverridesMonitor
 
 	mu struct {
 		syncutil.Mutex
 
+		dec       RowDecoder
```

Comment:
Looks like `pkg/sql/sem/tree.(*DatumAlloc)` is not: https://teamcity.cockroachdb.com/viewLog.html?buildId=8333460&tab=buildResultsDiv&buildTypeId=Cockroach_Ci_Tests_MaybeStressrace

Reply:
ah, right, forgot about the alloc. Perhaps the right thing to do is to decouple the alloc from the row decoder and pass it into the one method. We can have it allocate a new one if it's nil.

Reply:
Done, it's actually no longer needed because we do not use the same decoder but I thought it might be useful to make the decoder thread safe for the future.